src/main/java/org/GehtSoftHWByAziz/HW6/Task1/CustomExecutorService.java

```
@@ -0,0 +1,296 @@
+
+  @Override
+  public boolean awaitTermination(
+          long timeout,
+          TimeUnit timeUnit) throws InterruptedException {
+    long millis = timeUnit.toMillis(timeout);
+    long nanos = timeUnit.toNanos(timeout);
+    // used lock to make condition check and
+    if (useVirtualThreads) {
+      Thread.sleep(millis);
+      if (checkTermination()) {
+        return true;
+      } else {
+        return false;
+      }
+    }
+    lock.lock();
+    try {
+      while (!isTerminated) { // защита от ложного пробуждения
+//        A thread can also wake up without being notified, interrupted, or "
+//                timing out, a so-called spurious wakeup.
+        if (nanos <= 0) {
+          return false;
+        }
+        // if time is out
+        // returns value <=0
+        // if condition is met then isTerminated is set to true
+        nanos = terminationCondition.awaitNanos(nanos);
+      }
+      return true;
+    } finally {
+      lock.unlock();
+    }
+  }
+
+  @Override
+  public <T> Future<T> submit(Callable<T> callable) {
+    // RunnableFuture<T> implements RUNNABLE and FUTURE
+    RunnableFuture<T> ftask = new FutureTask<>(callable);
+
+    execute(ftask);
+    return ftask;
+  }
+
+  @Override
+  public <T> Future<T> submit(Runnable runnable, T t) {
+    RunnableFuture<T> ftask = new FutureTask<>(runnable, t);
+    execute(ftask);
+    return ftask;
+  }
+
+  @Override
+  public Future<?> submit(Runnable runnable) {
+    RunnableFuture<Void> ftask = new FutureTask<>(runnable, null);
+    execute(ftask);
+    return ftask;
+  }
+
+  @Override
+  public <T> List<Future<T>> invokeAll(
+          Collection<? extends Callable<T>> collection) throws InterruptedException {
+    List<Future<T>> tasks = new ArrayList<>();
+    for (var task : collection) {
+      var future = submit(task);
+      tasks.add(future);
+    }
+    for (var future : tasks) {
+      try {
+        future.get();
+      } catch (ExecutionException e) {
+
+      }
```

empty catch is a code smell, errors should never pass silently